lib/galaxy/config/parsers.py

```diff
@@ -0,0 +1,21 @@
+import ipaddress
+from typing import (
+    List,
+    Union,
+)
+
+from galaxy.util import unicodify
+
+IpAddressT = Union[ipaddress.IPv4Address, ipaddress.IPv6Address]
+IpNetworkT = Union[ipaddress.IPv4Network, ipaddress.IPv6Network]
+IpAllowedListEntryT = Union[IpAddressT, IpNetworkT]
+
+
+def parse_allowlist_ips(fetch_url_allowlist: List[str]) -> List[IpAllowedListEntryT]:
```

Comment:
I would put this in galaxy-util to really simplify the packages that are needed for galaxy-files. This dependency structure seems heavy handed to me.

Reply:
It does belong here though, doesn't it ? And file sources need config values, so this doesn't seem so outlandish that you'd bring in galaxy-config ?
The extra packages this brings in are
```
    boltons
    jinja2
    pykwalify
    PyYAML
```
afaict

Reply:
i've made the change but I think we will bring in more stuff that doesn't belong in util in the future.

Reply:
I don’t think it does belong there for it is worth. I think of the galaxy-config package as the huge app config loading process. I’d love for individual packages and components to have much simpler protocols for configuration that are isolated and don’t require all that heavy infrastructure and complex environment logic. I could totally be wrong though - maybe config is smaller than I thought or these packages are needed in util. I’ll happily approve the PR in either form.

Reply:
I guess we'll see how that pans out, package-specific config sounds great ... although this PR is probably a good example of how that'll be tricky 😆